server/webapp/WEB-INF/rails.new/app/controllers/admin/status_reports_controller.rb

```diff
@@ -32,16 +32,27 @@ def show
       render_plugin_error e
     end
 
+    def show
+      @view_title = 'Plugin Status Report'
+      @page_header = 'Plugin Status Report'
+      @agent_status_report = elastic_agent_extension.getAgentStatusReport(params[:plugin_id], params[:elastic_agent_id])
```

Comment:
What should be the behaviour when an agent is not assigned?

Reply:
@maheshp -- If the job is not assigned yet, it shows the *plugin status report* instead of *Agent Status Report* button. The check is done [here](https://github.com/GaneshSPatil/gocd/blob/7ec2e32c0010955951689b33a42e49474867748e/server/webapp/WEB-INF/vm/build_detail/_build_detail_summary_jstemplate.vm#L32-L36)

Going forward, we will add *Errors and Warnings* sections to the EA endpoint, which will tell errors regarding agent, job assignment failure, cluster, etc.

Reply:
Please correct me if I am wrong. I thought an agent status report would be more useful in cases where an agent is not yet assigned. I thought the plugin through the agent status report should be able to answer questions around the status of the agent creation, wouldn't this be possible since we send the job identifier to the plugin.